grails-app/services/au/org/ala/biocollect/merit/DocumentService.groovy

```diff
@@ -88,6 +91,112 @@ class DocumentService {
         return resp
     }
 
+    Map allDocumentsSearch(Integer offset = 0, Integer max = 100, String searchTerm = null, String searchType = null, String sort = null, String order = null, String projectId = null, String hub = null) {
+        String searchTextBy = "";
+
+        Map params = [:]
+
+        if (searchType == 'none' && searchTerm)
+            searchTextBy += "(name:" + searchTerm + " OR role:" + searchTerm + " OR labels:" + searchTerm + " OR attribution:" + searchTerm + " OR citation:" + searchTerm + " OR description:" + searchTerm + ")";
+
+        //projectId is passed in the case of viewing project documents
+        if (projectId) {
+            if (!searchTextBy.isEmpty())
+                searchTextBy += " AND projectId:" + projectId;
+            else
+                searchTextBy += "projectId:" + projectId;
+
+            if (searchType && searchTerm) {
+                if (searchType != 'none')
+                    searchTextBy += " AND " + searchType + ":" + searchTerm;
+            }
+
+            params = [
+                    offset: offset,
+                    max   : max,
+                    query : searchTextBy,
+                    fq    : DOCUMENT_FILTER
+            ]
+
+            Boolean isALAAdmin = userService.userIsAlaAdmin()
+            def members = projectService.getMembersForProjectId(projectId)
+
+            String userId = userService.getCurrentUserId()
+
+            boolean isViewable = false
+
+            if (userId)
+                isViewable = members.find{it.userId == userId} || isALAAdmin
```

Comment:
move `def members = projectService.getMembersForProjectId(projectId)` here . This will minimise calls to ecodata.